packages/flutter/lib/src/widgets/icon_theme_data.dart

```diff
@@ -100,16 +95,13 @@ class IconThemeData {
   int get hashCode => hashValues(color, opacity, size);
 
   @override
-  String toString() {
-    final List<String> result = <String>[];
-    if (color != null)
-      result.add('color: $color');
-    if (_opacity != null)
-      result.add('opacity: $_opacity');
-    if (size != null)
-      result.add('size: $size');
-    if (result.isEmpty)
-      return '<no theme>';
-    return result.join(', ');
+  void debugFillProperties(DiagnosticPropertiesBuilder description) {
+    super.debugFillProperties(description);
+    if (color == null && _opacity == null && size == null) {
+      return;
+    }
```

Comment:
we should always add the properties, just give them default values in the description so they get hidden by default in the output

Reply:
OK, I set them all to have default values of 'null'.